### PR TITLE
[Merged by Bors] - Initialize PoST with OpenCL

### LINKED ISSUE #130: Integration of OpenCL initialiser into `post` and `go-spacemesh`
### Description

The goal is to replace the existing `gpu-post` integration with the newly written OpenCL intitializer

### Acceptance Criteria

- Existing `gpu-post` integration is disabled, not removed (in case it is still needed). Switching between OpenCL and `gpu-post` is OK to require a rebuild of `post`.
- Test if go-spacemesh can initialize using OpenCL instead of `gpu-post`

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Closes https://github.com/spacemeshos/post/issues/130

Do not merge before https://github.com/spacemeshos/post/pull/131 and https://github.com/spacemeshos/api/pull/236

## Changes
This migrates PoST initialization to use `postrs` (OpenCL) instead of `gpu-post` (CUDA / Vulkan).

## Test Plan
New all existing tests related to initialization, proof generation and verification pass.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: activation/interface.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type atxProvider interface {
 // PostSetupProvider defines the functionality required for Post setup.
 type postSetupProvider interface {
 	Status() *PostSetupStatus
-	ComputeProviders() []PostSetupComputeProvider
+	ComputeProviders() ([]PostSetupComputeProvider, error)
```
Comment (fasmat) [inline activation/interface.go:59]: Rename to `Providers`

File: api/grpcserver/interface.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ type atxProvider interface {
 
 type postSetupProvider interface {
 	Status() *activation.PostSetupStatus
-	ComputeProviders() []activation.PostSetupComputeProvider
+	ComputeProviders() ([]activation.PostSetupComputeProvider, error)
```
Comment (fasmat) [inline api/grpcserver/interface.go:51]: Rename to `Providers`

File: api/grpcserver/smesher_service.go [hunk 3]
```diff
@@ -219,7 +222,7 @@ func (s SmesherService) PostSetupComputeProviders(ctx context.Context, in *pb.Po
 		res.Providers[i] = &pb.PostSetupComputeProvider{
 			Id:          uint32(p.ID),
 			Model:       p.Model,
-			ComputeApi:  pb.PostSetupComputeProvider_ComputeApiClass(p.ComputeAPI), // assuming enum values match.
+			ComputeApi:  pb.PostSetupComputeProvider_ComputeApiClass(p.DeviceType), // TODO(mafa): update API
```
Comment (fasmat) [inline api/grpcserver/smesher_service.go:225]: TODO: Update Protobuf in `api` repositor

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors cancel

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -106,6 +110,12 @@ jobs:
         uses: actions/setup-go@v4
         with:
           go-version: ${{ env.go-version }}
+      - name: Add OpenCL support for Linux
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: sudo apt-get update -q && sudo apt-get install -qy ocl-icd-opencl-dev libpocl2
+      - name: Add OpenCL support for Windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: choco install opencl-intel-cpu-runtime
```
Comment (poszu) [inline .github/workflows/ci.yml]: Is the runtime needed to **build**? 🤔 Perhaps it contains the required development files too 🤷 

Did you try `vcpkg install opencl` instead?

**Review by poszu — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**